components/cursor/cursor.hpp

```diff
@@ -21,16 +21,16 @@ namespace components::cursor {
 
     enum class operation_status_t { success = 1, failure = 0 };
 
-    enum class error_code_t {
+    enum class error_code_t : int32_t {
         none = 0,
-        database_already_exists,
-        database_not_exists,
-        collection_already_exists,
-        collection_not_exists,
-        collection_dropped,
-        sql_parse_error,
-        create_phisical_plan_error,
-        other_error
+        database_already_exists = 1,
+        database_not_exists = 2,
+        collection_already_exists = 3,
+        collection_not_exists = 4,
+        collection_dropped = 5,
+        sql_parse_error = 6,
+        create_phisical_plan_error = 7,
+        other_error = -1
```

Comment:
If you decided to numerate all enums, standard approach to place default and errors enums at the top. 
But in this case in my opinion use `uint16_t`  and for  `other_error = 1'000`